terminal/src/identity.rs

```diff
@@ -0,0 +1,63 @@
+use librad::profile::Profile;
+
+use radicle_common::fmt;
+use radicle_common::profile;
+
+use crate as term;
+
+/// Identity formatter that takes a profile and displays it `peer_id` and
+/// `username` depending on the configuration.
+///
+/// If short display is requested, `peer_id` is printed in its compact format
+/// e.g. `hynddpk…uf4qwge`. If `styled` is requested, `peer_id` and `username`
+/// are printed using the terminal's styled formatters.
+pub struct Formatter<'a> {
+    profile: &'a Profile,
+    shorten: bool,
+    style: bool,
+}
+
+impl<'a> Formatter<'a> {
+    pub fn new(profile: &'a Profile) -> Self {
+        Self {
+            profile,
+            shorten: false,
+            style: false,
+        }
+    }
+
+    pub fn shorten(mut self) -> Self {
+        self.shorten = true;
+        self
+    }
+
+    pub fn style(mut self) -> Self {
+        self.style = true;
+        self
+    }
+
+    pub fn print(self) -> String {
```

Comment:
Perhaps this can be moved to be an instance of `fmt::Display`, that way it doesn't have to be called explicitly.